python/pip_install/pip_repository.bzl

```diff
@@ -110,6 +110,8 @@ def _get_xcode_location_cflags(rctx):
 
     # Locate xcode-select
     xcode_select = rctx.which("xcode-select")
+    if not xcode_select.exists:
```

Comment:
According to the docs, which() returns None if the binary isn't found. Are you find otherwise? That it returns a path object, but exists is False?

https://bazel.build/rules/lib/builtins/repository_ctx#which

> Returns the path of the corresponding program or None if there is no such program in the path.

Reply:
I will update and test

Reply:
I created a function to handle all the which calls, and I am now testing for None.  I moved `uname` to `uname-hide` and it works like a champ :)